js/src/forum/addFlagsToPosts.js

```diff
@@ -20,15 +20,15 @@ export default function() {
 
     this.subtree.invalidate();
 
-    if (app.cache.flags) {
-      app.cache.flags.some((flag, i) => {
+    if (app.flags.cache) {
+      app.flags.cache.some((flag, i) => {
         if (flag.post() === post) {
-          app.cache.flags.splice(i, 1);
+          app.flags.cache.splice(i, 1);
 
           if (app.cache.flagIndex === post) {
-            let next = app.cache.flags[i];
+            let next = app.flags.cache[i];
 
-            if (!next) next = app.cache.flags[0];
+            if (!next) next = app.flags.cache[0];
```

Comment:
What about `app.cache.flagIndex` a few lines below? Should that become part of the state as well?